lib/Conversion/PolynomialToStandard/PolynomialToStandard.cpp

```diff
@@ -145,7 +145,7 @@ struct ConvertConstant : public OpConversionPattern<ConstantOp> {
     }
     for (const auto &term : attr.getPolynomial().getTerms()) {
       coeffs[term.exponent.getZExtValue()] = rewriter.getIntegerAttr(
-          eltTy, term.coefficient.trunc(eltTy.getIntOrFloatBitWidth()));
+          eltTy, term.coefficient.zextOrTrunc(eltTy.getIntOrFloatBitWidth()));
```

Comment:
hmm i'd definitely expect there to be tests that would detect if something weird is happening here, but is there an issue with zero extending the elements that are otherwise using signed arithmetic? (is the behavior different if you use `sextOrTrunc`?) I wonder if nothing is being detected because the coeff width is usually just large enough for the coefficients?

Reply:
Let me use `sextOrTrunc` and then I was going to follow up this PR with more tests, and I will try to exercise this one as well.